src/bookshelf/index.ts

```diff
@@ -30,25 +30,30 @@ export default class Bookshelf implements Mapper {
     let linkOpts: LinkOpts = { baseUrl: this.baseUrl, type, pag: bookOpts.pagination };
 
     // Set default values for the options
-    defaultsDeep(bookOpts, {relations: { included: true }, enableLinks: true, omitAttrs: []});
+    defaultsDeep(bookOpts, {
+      omitAttrs: [],
+      keyForAttr: identity,
+      relations: true,
+      typeForModel: (attr: string) => plural(attr),
+      enableLinks: true
+    });
 
     let info: Information = { bookOpts, linkOpts };
 
     let template: SerialOpts = processData(info, data);
 
-    let relationTypes: RelationTypeOpt = bookOpts.relationTypes || {};
-    function typeForAttribute(attr: string): string {
-      // if typeForAttribute returns undefined, the serializer
-      // omits the result and handles pluralization
-      if (typeof relationTypes === 'object') {
-        return relationTypes[attr];
-      } else {
-        return (relationTypes as RelationTypeFunction)(attr);
-      }
+    let { typeForModel, keyForAttr }: BookOpts = bookOpts;
+    let typeForAttribute: (attr: string) => string;
+
+    if (typeof typeForModel === 'function') {
+      typeForAttribute = typeForModel as ((attr: string) => string); // XXX remove `as ...` in typescript 2
+    } else {
+      // if the typeForModel object returns a falsy value, pluralize the attribute
+      typeForAttribute = (attr: string) =>  typeForModel[attr] || plural(attr);
```

Comment:
Curious if this is necessary given that the serialize pluralizes by default and has an option to specify so (`pluralizeType`). I know we're trying to prevent adverse affects from underlying serializer changes, but seems like we can avoid pluralizing twice.